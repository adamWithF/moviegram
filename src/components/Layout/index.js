import React from "react"
import propTypes from "prop-types"

import { Link, withRouter } from "react-router-dom"

import Search from "../../components/Search"

const Layout = ({ children, backLocation, searchHandler }) => (
  <div className="layout">
    <div className="header-main">
      {backLocation && (
        <Link className="backlink button" to={backLocation}>
          Back
        </Link>
      )}
      {searchHandler && <Search onSearch={searchHandler} />}
    </div>
    <main>{children}</main>
  </div>
)

Layout.propTypes = {
  children: propTypes.element.isRequired,
  backLocation: propTypes.string,
  searchHandler: propTypes.func
}

export default withRouter(Layout)
