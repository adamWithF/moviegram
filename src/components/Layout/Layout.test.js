import React from "react"
import { shallow } from "enzyme"

import Layout from "../Layout"

describe("Layout", () => {
  let props

  beforeEach(() => {
    props = {
      children: () => <div />,
      backLocation: undefined,
      searchHandler: undefined
    }
  })

  it("should render properly when backLocation provided and no searchHandler", () => {
    props.backLocation = "/"
    const wrapper = shallow(<Layout {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should render properly when searchHandler provided and no backLocation", () => {
    props.searchHandler = () => null
    const wrapper = shallow(<Layout {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
