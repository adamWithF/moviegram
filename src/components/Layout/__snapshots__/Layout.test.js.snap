// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render properly when backLocation provided and no searchHandler 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Layout)
    backLocation="/"
    searchHandler={undefined}
>
    [Function]
</withRouter(Layout)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Layout should render properly when searchHandler provided and no backLocation 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Layout)
    backLocation={undefined}
    searchHandler={[Function]}
>
    [Function]
</withRouter(Layout)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
